src/module.c

```diff
@@ -3801,6 +3801,9 @@ RedisModuleKey *RM_OpenKey(RedisModuleCtx *ctx, robj *keyname, int mode) {
     RedisModuleKey *kp;
     robj *value;
     int flags = 0;
+    if (mode & REDISMODULE_OPEN_KEY_NOEFFECTS) {
+        mode |= REDISMODULE_OPEN_KEY_NOTOUCH | REDISMODULE_OPEN_KEY_NONOTIFY | REDISMODULE_OPEN_KEY_NOSTATS | REDISMODULE_OPEN_KEY_NOEXPIRE;
```

Comment:
let's just set `flags` to `LOKKUP_NOEFFECTS`.
more compatible with other future changes in lookupKey.